packages/ui/src/Forms/CustomerProfile/CustomerProfileForm.tsx

```diff
@@ -0,0 +1,319 @@
+import React, { useState } from "react";
+import { Formik, Field, FieldProps } from "formik";
+import * as Yup from "yup";
+
+import { Customer } from "@eisbuk/shared";
+import i18n, {
+  useTranslation,
+  ValidationMessage,
+  CustomerLabel,
+  ActionButton,
+} from "@eisbuk/translations";
+import {
+  User,
+  Cake,
+  Mail,
+  Phone,
+  ClipboardList,
+  ShieldCheck,
+} from "@eisbuk/svg";
+
+import Button, { ButtonSize } from "../../Button";
+import TextInput, { IconAdornment } from "../../TextInput";
+import DateInput from "../../DateInput";
+import Checkbox from "../../Checkbox";
+
+import { isISODay } from "../../utils/date";
+import { isValidPhoneNumber } from "../../utils/helpers";
+
+interface FormProps {
+  customer: Partial<Customer>;
+  onCancel?: () => void;
+  onSave?: (customer: Customer) => void;
+}
+
+const defaultCustomerFormValues = {
+  name: "",
+  surname: "",
+  email: "",
+  phone: "",
+  birthday: "",
+  certificateExpiration: "",
+  covidCertificateReleaseDate: "",
+  covidCertificateSuspended: false,
+};
+
+const CustomerProfileForm: React.FC<FormProps> = ({
+  customer,
+  onCancel = () => {},
+  onSave = () => {},
+}) => {
+  const { t } = useTranslation();
+  const [isEditing, setIsEditing] = useState(false);
+
+  const toggleEdit = () => setIsEditing(!isEditing);
+
+  const initialValues = {
+    ...defaultCustomerFormValues,
+    ...customer,
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={CustomerValidation}
+      onSubmit={(values, { setSubmitting }) => {
+        onSave(values as Customer);
+        setSubmitting(false);
+        toggleEdit();
+      }}
+    >
+      {({ resetForm, handleSubmit, isSubmitting }) => (
+        <form onSubmit={handleSubmit}>
```

Comment:
Instead of this you can use Formik's `Form` component. It renders a `form` and accepts `submit` and `reset` functionality (from button children), but it doesn't need `onSubmit` as it connects to `Formik` through context